apps/passport-client/new-components/screens/Home/NewHomeScreen.tsx

```diff
@@ -0,0 +1,456 @@
+import { ChevronLeftIcon, ChevronRightIcon } from "@heroicons/react/24/solid";
+import {
+  EdDSATicketPCDTypeName,
+  ITicketData,
+  isEdDSATicketPCD
+} from "@pcd/eddsa-ticket-pcd";
+import { Spacer } from "@pcd/passport-ui";
+import { PCD } from "@pcd/pcd-types";
+import { PODTicketPCDTypeName, isPODTicketPCD } from "@pcd/pod-ticket-pcd";
+import {
+  ReactElement,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState
+} from "react";
+import { useNavigate } from "react-router-dom";
+import styled, { FlattenSimpleInterpolation, css } from "styled-components";
+import { AppContainer } from "../../../components/shared/AppContainer";
+import { CardBody } from "../../../components/shared/PCDCard";
+import { useDispatch, usePCDs, useSelf } from "../../../src/appHooks";
+import { useSyncE2EEStorage } from "../../../src/useSyncE2EEStorage";
+import { FloatingMenu } from "../../shared/FloatingMenu";
+import { NewModals } from "../../shared/Modals/NewModals";
+import { TicketCard } from "../../shared/TicketCard";
+import { Typography } from "../../shared/Typography";
+import { TicketPack, TicketType, TicketTypeName } from "./types";
+import { AddOnsModal } from "./AddOnModal";
+
+const EVENT_GAP = 4;
+const TICKETS_HORIZONTAL_GAP = 40;
+const SHOW_HELPER_LINES = false;
+const TICKET_VERTICAL_GAP = 20;
+
+const isEventTicketPCD = (pcd: PCD<unknown, unknown>): pcd is TicketType => {
+  // TODO: fetch the pods type as well and prioritize it if theres a conflict.
+  return isEdDSATicketPCD(pcd) || isPODTicketPCD(pcd);
+};
+
+const useTickets = (): Array<[string, TicketPack[]]> => {
+  const allPCDs = usePCDs();
+  const tickets = allPCDs.filter(isEventTicketPCD);
+  const ticketsTrigger = tickets.map((t) => t.id).join(" ");
+  return useMemo(() => {
+    const eventsMap = new Map<string, TicketPack[]>();
+    for (const ticket of tickets) {
+      if (ticket.claim.ticket.ticketName !== "GA") continue;
+      let ticketPacks = eventsMap.get(ticket.claim.ticket.eventName);
+      if (!ticketPacks) {
+        ticketPacks = [];
+        eventsMap.set(ticket.claim.ticket.eventName, ticketPacks);
+      }
+      if (ticket.type === PODTicketPCDTypeName) {
+        const relatedEddesaTicketPackIdx = ticketPacks.findIndex(
+          (pack) =>
+            pack.attendeeEmail === ticket.claim.ticket.attendeeEmail &&
+            pack.packType === EdDSATicketPCDTypeName
+        );
+        if (relatedEddesaTicketPackIdx >= 0)
+          ticketPacks.splice(relatedEddesaTicketPackIdx, 1);
+      }
+      ticketPacks.push({
+        eventTicket: ticket,
+        eventId: ticket.claim.ticket.eventId,
+        addOns: [],
+        attendeeEmail: ticket.claim.ticket.attendeeEmail,
+        packType: ticket.type as TicketTypeName
+      });
+    }
+    for (const ticket of tickets) {
+      if (ticket.claim.ticket.ticketName === "GA") continue;
```

Comment:
i'm not understanding this `GA` special-casing, could you please leave a comment explaining it?

Reply:
https://linear.app/0xparc-pcd/issue/0XP-1270/more-correct-implementation-of-addon-item-groupings